test/nl/surf/eduhub_rio_mapper/e2e_helper.clj

```diff
@@ -0,0 +1,490 @@
+                 uuid)
+               id)])))
+
+(defn- call-api
+  "Make API call, print results and http-message, and return response."
+  [method action args]
+  (let [args          (if (= method :post)
+                        (interpret-post-job-args args)
+                        args)
+        url           (str @base-url (api-path action args))
+        req           {:method           method
+                       :url              url
+                       :headers          {"Authorization" (str "Bearer "
+                                                               @bearer-token)}
+                       :query-params     {:http-messages "true"}
+                       :as               :json
+                       :throw-exceptions false}
+        res           (http/request req)
+        http-messages (-> res :body :http-messages)
+        res           (if (map? (:body res)) ;; expect JSON response
+                        ;; but can be something
+                        ;; else on error
+                        (update res :body dissoc :http-messages)
+                        res)]
+    (print-api-message {:req req, :res res})
+    (when (seq http-messages)
+      (print-boxed "Job HTTP messages"
+        (print-http-messages http-messages)))
+    res))
+
+(defn- api-status-final?
+  "Determine if polling can be stopped from API status call response."
+  [res]
+  (cli/final-status? (-> res :body :status keyword)))
+
+(def job-status-poll-sleep-msecs 500)
+(def job-status-poll-total-msecs 60000)
+
+(defn post-job
+  "Post a job through the API.
+  Return the HTTP response of the call and includes a \"delay\" to access
+  the job result at `:result-delay`."
+  [action & args]
+  (let [{:keys [status] {:keys [token]} :body :as res}
+        (call-api :post action args)]
+    (assoc res :result-delay
+           (delay
+             (if (= http-status/ok status)
+               (loop [tries-left (/ job-status-poll-total-msecs
+                                    job-status-poll-sleep-msecs)]
+                 (let [{:keys [status body] :as res}
+                       (call-api :get :status [token])]
+                   (cond
+                     (zero? tries-left)
+                     (do
+                       (println "\n\n⚠ too many retries on status\n")
+                       ::time-out)
+
+                     (not= http-status/ok status)
+                     (do
+                       (println "\n\n⚠ get status failed\n")
+                       ::get-status-failed)
+
+                     (api-status-final? res)
+                     body
+
+                     :else
+                     (do
+                       (Thread/sleep job-status-poll-sleep-msecs)
+                       (recur (dec tries-left))))))
+               (println "failed to post job"))))))
+
+(defn job-result
+  "Use `get-in` to access job response from `post-job`."
+  [job & ks]
+  (get-in @(:result-delay job) ks))
+
```

Comment:
I think getting rid of the `job-` prefixes for these functions looks nicer in the tests, but it's not blocking the PR for me